ethcore/src/state/mod.rs

```diff
@@ -47,14 +47,16 @@ pub type ApplyResult = Result<ApplyOutcome, Error>;
 /// Account modification state. Used to check if the account was
 /// Modified in between commits and overall.
 enum AccountState {
-	/// Account was never modified in this state object.
-	Clean,
+	/// Account was loaded from disk and never modified in this state object.
+	CleanFresh,
+	/// Account was loaded from the global cache and never modified.
+	CleanCached,
```

Comment:
This tracks accounts that were loaded from the global cache to be modified, but modification was reverted with the snapshot. In such case they won't be passed back to the global cache.